include/command_history.hpp

```diff
@@ -0,0 +1,31 @@
+/**
+ * Project: catan
+ * @file common.hpp
+ * @brief common data types
+ *
+ * @author Bowei Li <cesterty024@gmail.com>
+ *
+ * All right reserved.
+ */
+
+#ifndef INCLUDE_COMMAND_HISTORY_HPP
+#define INCLUDE_COMMAND_HISTORY_HPP
+
+#include<deque>
+#include<string>
+
+class CommandHistory
+{
+private:
+    std::deque<std::string>mHistory;
+    size_t mCurrentIndex = 0;
+    std::string mCurrentCommand;
```

Comment:
seems to me, `mCurrentCommand` will always be `mHistory.at(mCurrentIndex)`
thus, no need for this variable